src/contracts/core/DelegationManager.sol

```diff
@@ -341,239 +406,245 @@ contract DelegationManager is
     }
 
     /**
-     * @dev commented-out param (middlewareTimesIndex) is the index in the operator that the staker who triggered the withdrawal was delegated to's middleware times array
-     * This param is intended to be passed on to the Slasher contract, but is unused in the M2 release of these contracts, and is thus commented-out.
+     * @dev This function completes a queued withdrawal for a staker.
+     * This will apply any slashing that has occurred since the the withdrawal was queued by multiplying the withdrawal's
+     * scaledShares by the operator's maxMagnitude for each strategy. This ensures that any slashing that has occurred
+     * during the period the withdrawal was queued until its completable timestamp is applied to the withdrawal amount.
+     * If receiveAsTokens is true, then these shares will be withdrawn as tokens.
+     * If receiveAsTokens is false, then they will be redeposited according to the current operator the staker is delegated to,
+     * and added back to the operator's delegatedShares.
      */
     function _completeQueuedWithdrawal(
-        Withdrawal calldata withdrawal,
+        Withdrawal memory withdrawal,
         IERC20[] calldata tokens,
-        uint256, /*middlewareTimesIndex*/
         bool receiveAsTokens
     ) internal {
+        require(tokens.length == withdrawal.strategies.length, InputArrayLengthMismatch());
+        require(msg.sender == withdrawal.withdrawer, WithdrawerNotCaller());
         bytes32 withdrawalRoot = calculateWithdrawalRoot(withdrawal);
+        require(pendingWithdrawals[withdrawalRoot], WithdrawalNotQueued());
+
+        uint256[] memory prevSlashingFactors;
+        {
+            uint32 completableBlock = withdrawal.startBlock + MIN_WITHDRAWAL_DELAY_BLOCKS;
+            require(completableBlock <= uint32(block.number), WithdrawalDelayNotElapsed());
+
+            // Given the max magnitudes of the operator the staker was originally delegated to, calculate
+            // the slashing factors for each of the withdrawal's strategies.
+            prevSlashingFactors = _getSlashingFactorsAtBlock({
+                staker: withdrawal.staker,
+                operator: withdrawal.delegatedTo,
+                strategies: withdrawal.strategies,
+                blockNumber: completableBlock
+            });
+        }
 
-        require(pendingWithdrawals[withdrawalRoot], WithdrawalDoesNotExist());
-
-        require(withdrawal.startBlock + minWithdrawalDelayBlocks <= block.number, WithdrawalDelayNotElapsed());
-
-        require(msg.sender == withdrawal.withdrawer, UnauthorizedCaller());
+        // Given the max magnitudes of the operator the staker is now delegated to, calculate the current
+        // slashing factors to apply to each withdrawal if it is received as shares.
+        address newOperator = delegatedTo[withdrawal.staker];
+        uint256[] memory newSlashingFactors = _getSlashingFactors(withdrawal.staker, newOperator, withdrawal.strategies);
```

Comment:
bit of an optimization nit, but I think this can be moved inside the `if...else` block where `receiveAsToken = false`